nixos/modules/services/mail/postsrsd.nix

```diff
@@ -107,29 +90,42 @@ in
       postsrsd.gid = config.ids.gids.postsrsd;
     };
 
+    systemd.services.postsrsd-generate-secrets = {
+      path = [ pkgs.coreutils ];
+      script = ''
+        if [ -e "${cfg.secretsFile}" ]; then
+          echo "Secrets file exists. Nothing to do!"
+        else
+          echo "WARNING: secrets file not found, autogenerating!"
+          DIR="$(dirname "${cfg.secretsFile}")"
+          install -m 750 -o ${cfg.user} -g ${cfg.group} -d "$DIR"
+          install -m 600 -o ${cfg.user} -g ${cfg.group} <(dd if=/dev/random bs=18 count=1 | base64) "${cfg.secretsFile}"
+        fi
+      '';
+      serviceConfig = {
+        Type = "oneshot";
+      };
+    };
+
     systemd.services.postsrsd = {
       description = "PostSRSd SRS rewriting server";
-      after = [ "network.target" ];
+      after = [
+        "network.target"
+        "postsrsd-generate-secrets.service"
+      ];
```

Comment:
```suggestion
      after = [ "network.target" ];
```

Reply:
Marking this resolved, as per discussion [here](https://github.com/NixOS/nixpkgs/pull/397316#discussion_r2040689964).